### PR TITLE
 Fix Spark 3.2.1 build failures and make it non-snapshot

### PR DESCRIPTION
**Pull request by tgravescs**
Spark has 3.2.1 up for vote, I was testing and found it wouldn't build.  The issue here is that https://issues.apache.org/jira/browse/SPARK-37705 was merged in but our internal build didn't pick it up due to spark haven't already changed the version number to 3.2.2-snapshot.

 This is WIP because it has temporary repository addition to build against the 3.2.1 rc1 version.  Once it's officially released I will update this to remove that repository. Putting up to run through build and get any comments early.

fixes https://github.com/NVIDIA/spark-rapids/issues/4500

### CODE REVIEW

File: dist/pom.xml [hunk 1]
```diff
@@ -182,6 +182,12 @@
             <properties>
                 <included_buildvers>${buildver}</included_buildvers>
             </properties>
+            <pluginRepositories>
```
Comment (tgravescs) [inline dist/pom.xml:185]: will be removed

File: jenkins/settings.xml [hunk 2]
```diff
@@ -36,6 +36,14 @@
   <profiles>
     <profile>
       <repositories>
+        <repository>
```
Comment (tgravescs) [inline jenkins/settings.xml:39]: will be removed

File: pom.xml [hunk 3]
```diff
@@ -1371,9 +1371,9 @@
     <repositories>
         <repository>
             <id>snapshots-repo</id>
-            <url>https://oss.sonatype.org/content/repositories/snapshots</url>
+            <url>https://repository.apache.org/content/repositories/orgapachespark-1395/</url>
```
Comment (tgravescs) [inline pom.xml:1374]: will be removed

**Comment by tgravescs** [comment]
build

**Review by jlowe — COMMENTED** [review]
lgtm once the temp repo changes are removed.

**Comment by tgravescs** [comment]
build

**Comment by tgravescs** [comment]
build

**Comment by tgravescs** [comment]
premerged timed out for some reason, looking

**Comment by tgravescs** [comment]
saw some pretty big pauses for now apparent reason. Just going to rekick and see what happens:

[2022-01-12T20:07:30.312Z] ^[[36mDiscovery starting.^[[0m
[2022-01-12T20:08:12.503Z] [gw2]^[[36m [ 94%] ^[[0m^[[32mPASSED^[[0m ../../src/main/python/udf_test.py::test_cogroup_apply_udf[Short(not_null)][IGNORE_ORDER, ALLOW_NON_GPU(FlatMapCoGroupsInPandasExec,PythonUDF,Alias)]
[2022-01-12T20:08:38.034Z] ^[[36mDiscovery completed in 1 minute.^[[0m

**Comment by tgravescs** [comment]
build

**Comment by tgravescs** [comment]
build

**Comment by tgravescs** [comment]
3.2.1 is released, I upmerged and am running the tests

**Review by jlowe — APPROVED** [review]

**Review by revans2 — APPROVED** [review]

**Pull request merged**